src/vis/bar/utils.ts

```diff
@@ -31,30 +35,118 @@ const defaultConfig: IBarConfig = {
+  const categoricalOptions = [...new Set(catColValues.map((v) => v.val))];
+
+  const categoricalMap = {};
+
+  catColValues.forEach((val) => {
+    categoricalMap[val.id] = val.val;
+  });
+
+  if (aggType === EAggregateTypes.COUNT) {
+    return categoricalOptions.map((curr) => (catColValues as VisCategoricalValue[]).filter((c) => c.val === curr).length);
+  }
+  if (aggType === EAggregateTypes.AVG) {
+    return categoricalOptions.map((curr) =>
+      mean((aggColValues as VisNumericalValue[]).filter((c) => categoricalMap[c.id] === curr && !Number.isNaN(c.val)).map((c) => c.val)),
+    );
+  }
+  if (aggType === EAggregateTypes.MIN) {
+    return categoricalOptions.map((curr) =>
+      min((aggColValues as VisNumericalValue[]).filter((c) => categoricalMap[c.id] === curr && !Number.isNaN(c.val)).map((c) => c.val)),
+    );
+  }
+  if (aggType === EAggregateTypes.MED) {
+    return categoricalOptions.map((curr) =>
+      median((aggColValues as VisNumericalValue[]).filter((c) => categoricalMap[c.id] === curr && !Number.isNaN(c.val)).map((c) => c.val)),
+    );
+  }
+
+  return categoricalOptions.map((curr) =>
+    max((aggColValues as VisNumericalValue[]).filter((c) => categoricalMap[c.id] === curr && !Number.isNaN(c.val)).map((c) => c.val)),
+  );
+}
+
+function getTotalAggregateValues(
+  aggType: EAggregateTypes,
+  categoricalOptions: string[],
+  catColValues: VisCategoricalValue[],
+  aggColValues: VisNumericalValue[],
+) {
+  const categoricalMap = {};
+
+  catColValues.forEach((val) => {
+    categoricalMap[val.id] = val.val;
+  });
+
+  if (aggType === EAggregateTypes.COUNT) {
+    return sum(categoricalOptions.map((curr) => (catColValues as VisCategoricalValue[]).filter((c) => c.val === curr).length));
+  }
+  if (aggType === EAggregateTypes.AVG) {
+    return sum(
+      categoricalOptions.map(
+        (curr) => mean((aggColValues as VisNumericalValue[]).filter((c) => categoricalMap[c.id] === curr && !Number.isNaN(c.val)).map((c) => c.val)) || 0,
+      ),
+    );
+  }
+  if (aggType === EAggregateTypes.MIN) {
+    return sum(
+      categoricalOptions.map(
+        (curr) => min((aggColValues as VisNumericalValue[]).filter((c) => categoricalMap[c.id] === curr && !Number.isNaN(c.val)).map((c) => c.val)) || 0,
+      ),
+    );
+  }
+  if (aggType === EAggregateTypes.MED) {
+    return sum(
+      categoricalOptions.map(
+        (curr) => median((aggColValues as VisNumericalValue[]).filter((c) => categoricalMap[c.id] === curr && !Number.isNaN(c.val)).map((c) => c.val)) || 0,
+      ),
+    );
+  }
+
+  return sum(
+    categoricalOptions.map(
+      (curr) => max((aggColValues as VisNumericalValue[]).filter((c) => categoricalMap[c.id] === curr && !Number.isNaN(c.val)).map((c) => c.val)) || 0,
+    ),
+  );
+}
+
 async function setPlotsWithGroupsAndMultiples(
   columns: VisColumn[],
   catCol: VisCategoricalColumn,
+  aggType: EAggregateTypes,
+  aggregateColumn: VisNumericalColumn,
```

Comment:
```suggestion
  aggregateType: EAggregateTypes,
  aggregateColumn: VisNumericalColumn,
```

Why one is abbreviated and one not?